msal/src/main/java/com/microsoft/identity/client/internal/controllers/MSALControllerFactory.java

```diff
@@ -125,12 +144,42 @@ public static boolean brokerEligible(@NonNull final Context applicationContext,
             return false;
         }
 
-        // Use broker if installed and verified
-        if (brokerInstalled(applicationContext)) {
-            return true;
+        // Check if broker installed
+        if (!brokerInstalled(applicationContext)) {
+            return false;
         }
 
-        return false;
+        // Check if MicrosoftAuthService supported or AccountManager permission granted
+        if (BrokerMsalController.isMicrosoftAuthServiceSupported(applicationContext)
```

Comment:
I'd definitely consider adding some logging here so that we know what happened in this case relative to the broker connection.